app/models/computer.rb

```diff
@@ -4,9 +4,16 @@ class Computer < ActiveRecord::Base
 
   belongs_to :country, required: false
   has_one :package, as: :detail, dependent: :destroy
-  after_commit :create_on_stockit, on: :create
-  after_update :update_on_stockit
   before_save :save_correct_country, if: :request_from_stockit?
   before_save :downcase_brand, if: :brand_changed?
   before_save :set_updated_by
+  before_save :map_drop_down_attributes, if: :request_from_stockit?
+  after_commit :create_on_stockit, on: :create
+  after_update :update_on_stockit
+
+  private
+
+  def map_drop_down_attributes
```

Comment:
@abhinav098 there should be no need to map the attributes. use the id of the lookup object. So 

`comp_test_status_id = lookup_id`

I may have confused you with the key/label stuff.
We just need standard rails `reference_id` tooling here.